lua/which-key/mappings.lua

```diff
@@ -0,0 +1,221 @@
+local Util = require("which-key.util")
+
+local M = {}
+
+local function lookup(...)
+  local ret = {}
+  for _, t in ipairs({ ... }) do
+    for _, v in ipairs(t) do
+      ret[v] = v
+    end
+  end
+  return ret
+end
+
+local mapargs = {
+  "noremap",
+  "desc",
+  "expr",
+  "silent",
+  "nowait",
+  "script",
+  "unique",
+  "callback",
+  "replace_keycodes", -- TODO: add config setting for default value
+}
+local wkargs = {
+  "prefix",
+  "mode",
+  "plugin",
+  "buffer",
+  "remap",
+  "cmd",
+  "name",
+  "group",
+  "preset",
+  "cond",
+}
+local transargs = lookup({
+  "noremap",
+  "expr",
+  "silent",
+  "nowait",
+  "script",
+  "unique",
+  "prefix",
+  "mode",
+  "buffer",
+  "preset",
+  "replace_keycodes",
+})
+local args = lookup(mapargs, wkargs)
+
+function M.child_opts(opts)
+  local ret = {}
+  for k, v in pairs(opts) do
+    if transargs[k] then
+      ret[k] = v
+    end
+  end
+  return ret
+end
+
+function M._process(value, opts)
+  local list = {}
+  local children = {}
+  for k, v in pairs(value) do
+    if type(k) == "number" then
+      if type(v) == "table" then
+        -- nested child, without key
+        table.insert(children, v)
+      else
+        -- list value
+        table.insert(list, v)
+      end
+    elseif args[k] then
+      -- option
+      opts[k] = v
+    else
+      -- nested child, with key
+      children[k] = v
+    end
+  end
+  return list, children
+end
+
+function M._parse(value, mappings, opts)
+  if type(value) ~= "table" then
+    value = { value }
+  end
+
+  local list, children = M._process(value, opts)
+
+  if opts.plugin then
+    opts.group = true
+  end
+  if opts.name then
+    -- remove + from group names
+    opts.name = opts.name and opts.name:gsub("^%+", "")
+    opts.group = true
+  end
+
+  -- fix remap
+  if opts.remap then
+    opts.noremap = not opts.remap
+    opts.remap = nil
+  end
+
+  -- fix buffer
+  if opts.buffer == 0 then
+    opts.buffer = vim.api.nvim_get_current_buf()
+  end
+
+  if opts.cond ~= nil then
+    if type(opts.cond) == "function" then
+      if not opts.cond() then
+        return
+      end
+    elseif not opts.cond then
+      return
+    end
+  end
+
+  -- process any array child mappings
+  for k, v in pairs(children) do
+    local o = M.child_opts(opts)
+    if type(k) == "string" then
+      o.prefix = (o.prefix or "") .. k
+    end
+    M._try_parse(v, mappings, o)
+  end
+
+  -- { desc }
+  if #list == 1 then
+    assert(type(list[1]) == "string", "Invalid mapping for " .. vim.inspect({ value = value, opts = opts }))
```

Comment:
I'm getting this error on startup. Any ideas what might cause this?

Reply:
```
Error detected while processing VimEnter Autocommands for "*":
...k/packer/start/which-key.nvim/lua/which-key/mappings.lua:134: Invalid mapping for {
  opts = {
    mode = "i",
    prefix = "<c-j>"
  },
  value = { false }
```

Reply:
Can you open an issue and post your full which-key config?